R/ppm.R

```diff
@@ -0,0 +1,13 @@
+#' PPM - Parts per Million
+#'
+#' `ppm` is a small helper function to mulitply *ppm* values with `1e6`.
+#'
+#' @param x `numeric`, ppm values.
+#'
+#' @return `numeric`: `x * 1e6`
+#'
+#' @author Sebastian Gibb
+#' @export
+#' @examples
+#' ppm(5)
+ppm <- function(x)x * 1e6
```

Comment:
I like the idea of the `ppm` function - that's something I need a lot. But first you have a mistake here, it's either `x * 1e-6` or `x / 1e6` and secondly, I would define it differently:

```r
ppm <- function(x, ppm = 0) x * ppm / 1e6
```

So users could get the ppm for their values directly. In fact, if you ask for the *ppm* for the values `x` it will return them. For the `closest` call above it would then be: `closest(x, y, tolerance = ppm(y, 20))`.

Reply:
Stupid me! Indeed its `x * 1e-6`. Fixed. BTW: that's exactly the reason why I need that function!

Regarding your second suggestion: I somehow like the `x * ppm(5)` idea but to be honest it is just useful for absolute errors (and your solution may avoid `x / ppm(5)` vs `x * ppm(5)` errors). How does this look like (I don't like the `ppm(1, 20)` here but is an *absolute* tolerance often used?)?:

| | absolute | relative |
|---|---|---|
|mine| `closest(x, y, tol = ppm(20))`|`closest(x, y, tol = y * ppm(20))`|
|yours|`closest(x, y, tol = ppm(1, 20))`|`closest(x, y, tol = ppm(y, 20))`|

Reply:
Honestly, I've never used the ppm in terms of absolute error - I'm always considering it as relative to the provided m/z. In that sense the parts-per-million makes also more sense to me, it's somehow like a percentage of something, so I would never use it without the *something* (which in our case is the m/z). AFAIK the ppm for MS instruments are also relative to the measured m/z - or am I wrong here?

I would like it to be changed.

Reply:
ok, done!